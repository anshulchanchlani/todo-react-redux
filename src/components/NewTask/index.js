import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addTask } from '../../actions'
import './index.scss'
class NewTask extends Component {
    constructor(props) {
        super(props)
        this.state = {
            taskName: '',
            error:false
        }
        this.taskStitchAndAdd = this.taskStitchAndAdd.bind(this)
        this.handleOnChange = this.handleOnChange.bind(this)
        this.handleOnKeyDown = this.handleOnKeyDown.bind(this)
    }
    taskStitchAndAdd() {
        if (this.state.taskName !== '') {
            let newTask = {
                id: this.props.tasks.length + 1,
                name: this.state.taskName.trim(),
                createdAt: Date.now(),
                completed: false,
                lastModifiedAt: Date.now()
            }

            this.props.addTask(Object.assign({}, newTask));
            this.props.saveToDisk();
            this.setState({ taskName: '' })
        }else{
            this.setState({error:true})
        }
    }
    handleOnChange(e) {
        this.setState({error:false})
        if (e.keyCode === 13) {
            this.taskStitchAndAdd()

        }
        this.setState({ taskName: e.target.value })
    }
    handleOnKeyDown(e) {

        if (e.keyCode === 13) {
            this.taskStitchAndAdd();
        }
    }
    render() {

        return (
            <Fragment>
            <section className="new-task-form">
                <input placeholder="Enter a task" value={this.state.taskName}
                    onChange={this.handleOnChange} onKeyDown={this.handleOnKeyDown} />
                    
                <button onClick={this.taskStitchAndAdd}>Add</button>
                
            </section>
            {this.state.error?<span className="error-msg">Please enter a valid value</span>:null}
            </Fragment>
        );
    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({ addTask }, dispatch)
}

function mapStateToProps(state) {
    return {
        tasks: state.tasks
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(NewTask)