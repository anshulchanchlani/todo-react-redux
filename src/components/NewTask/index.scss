.new-task-form{
    display:flex;
    justify-content: center;
    align-content: space-around;
   
}

.new-task-form>button{
    width:100px;
    height:25px;
    margin-left:5px;
}

.new-task-form>input{
    width:60%;
    height:20px;
    margin-left:60px;
    text-align: center;
}

.error-msg{
    margin-top:10px;
    display:flex;
    justify-content:center;
    align-content: space-around;
    color:red;
}