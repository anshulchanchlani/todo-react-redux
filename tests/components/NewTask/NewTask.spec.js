import React from 'react';
import NewTask from '../../../src/components/NewTask';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import {shallow} from 'enzyme';



describe('New Task tests',()=>{
    const wrapper = shallow(<NewTask/>); 
    const component = toJson(wrapper)
    
    test('NewTask renders',()=>{
      expect(toJson(wrapper)).toMatchSnapshot();
    })
})
