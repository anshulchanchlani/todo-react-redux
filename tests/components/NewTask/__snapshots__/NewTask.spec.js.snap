// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Task tests NewTask renders 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
